tree/ntuple/v7/src/RPageStorage.cxx

```diff
@@ -207,3 +234,33 @@ void ROOT::Experimental::Detail::RPageSink::CommitCluster(ROOT::Experimental::NT
    ++fLastClusterId;
    fPrevClusterNEntries = nEntries;
 }
+
+
+ROOT::Experimental::Detail::RPageStorage::RSealedPage
+ROOT::Experimental::Detail::RPageSink::SealPage(
+   const RPage &page, const RColumnElementBase &element, int compressionSetting)
+{
+   unsigned char *buffer = reinterpret_cast<unsigned char *>(page.GetBuffer());
+   bool isAdoptedBuffer = true;
+   auto packedBytes = page.GetSize();
+
+   if (!element.IsMappable()) {
+      packedBytes = element.GetPackedSize(page.GetNElements());
+      buffer = new unsigned char[packedBytes];
+      isAdoptedBuffer = false;
+      element.Pack(buffer, page.GetBuffer(), page.GetNElements());
+   }
+   auto zippedBytes = packedBytes;
+
+   if ((compressionSetting != 0) || !element.IsMappable()) {
+      zippedBytes = fCompressor->Zip(buffer, packedBytes, fOptions.GetCompression());
+      if (!isAdoptedBuffer)
```

Comment:
A similar feature might be useful for the Unsealing case (to reuse a scratch buffer rather than always reallocate it).

Reply:
That's a good point!  I added a TODO in `UnsealPage()`